src/library/scala/collection/mutable/TreeMap.scala

```diff
@@ -52,6 +52,20 @@ sealed class TreeMap[A, B] private (tree: RB.Tree[A, B])(implicit val ordering:
   override def empty = TreeMap.empty
   override protected[this] def newBuilder = TreeMap.newBuilder[A, B]
 
+  /**
+   * Creates a ranged projection of this map. Any mutations in the ranged projection will update the original map and
+   * vice versa.
+   *
+   * Only entries with keys between this projection's key range will ever appear as elements of this map, independently
+   * of whether the entries are added through the original map or through this view. That means that if one inserts a
```

Comment:
It's pretty weird to be able to add elements that you can't see.  Are we sure this is the right behavior?